### ISSUE TITLE
Unable to batch and COPY .parquet files via manifest file

### ISSUE THREAD

**Issue by andreys70** [awslabs/aws-lambda-redshift-loader#223]
Hi, 
I can't get this solution to work for .parquet files via manifest file. An each entry require "content_length" to be specified in manifest file, in order for COPY command execute successfully. While building manifest file in createManifest(), individual entry content_length is not properly populated, as it using s3Info.size (line 845) for it. So, each entry in manifest file will have the same size, which is incorrect. 
The attempt to COPY to Redshift produces the following error: 
"Error: HTTP response error code: 416 Message: InvalidRange The requested range is not satisfiable"

The solution is working only if there is one .parquet file in a batch/manifest file.

**Comment by IanMeyers** [comment]
Looking

**Comment by IanMeyers** [comment]
This will be fixed in 2.7.9. Just need to test it

**Comment by andreys70** [comment]
Ian, if code is ready I can help to test it. I'm really looking forward using this solution in our account.

**Comment by IanMeyers** [comment]
OK - it's not on NPM yet, but if you can download 2.7.9 from https://awslabs-code-us-east-1.s3.amazonaws.com/LambdaRedshiftLoader/AWSLambdaRedshiftLoader-2.7.9.zip and deploy it, you should see that there are 2 fundamental changes:

1. The file size is captured in the pending batch entry
2. The content length is set from this value on manifest creation

I've not yet tested backward compatibility for batches that already have files in them, but it should work - that is next. Lemme know...

**Comment by andreys70** [comment]
I'll give it a try. I don't care about previously created batches, this is new implementation, so more room for experiment.

**Comment by andreys70** [comment]
Ian, one other minor thing. The environment variable "AWS_REGION" can't be created in Lambda using web interface:
"Lambda was unable to configure your environment variables because the environment variables you have provided contains reserved keys that are currently not supported for modification. Reserved keys used in this request: AWS_REGION"

**Comment by IanMeyers** [comment]
Yes, that is fine. It will automatically pick up this variable from the region in which the function is running.

**Comment by andreys70** [comment]
Ian, I was running Kinesis Firehose data pipeline for past 4 hours to test various configuration for parquet file load. It processed ~300 files (~120MM records) for following batchSize/batchTiemoutSecs configurations: 2/120, 4/120, 4/240, 12/900, 24/1800.  I see no issues for version 2.7.9, all works as expected.

**Comment by IanMeyers** [comment]
That's great. I've got a new version of 2.7.9 that is backward compatible with 2.7.8, which is now up at the link posted above. I've tested and it works well, but would invite you to use this version and let me know if you see anything strange. Please note that it is NOT compatible with the previous version of 2.7.9 that you downloaded.

**Comment by IanMeyers** [comment]
Version 2.7.9 has been committed in 9443d92 and published to npm. Please let me know if you experience any issues.

**Comment by andreys70** [comment]
Thank you, I'll give it a try and let you know later today.

**Comment by andreys70** [comment]
Tested new deployment of 2.7.9, all works as expected - no issues. I've deployed this solution to production which COPY data to 2 Redshift clusters. This is very busy data pipeline, and LambdaRedshiftLoader performs very well. Also, it eliminates issues with Redshift table locking we had while COPY command executed at the very hight rate against same table (this was a mess). I think I can claim first production implementation of LamdbaRedshiftLoader supporting .parquet files. :) Thank you very much for a quick resolution, Ian! Looking forward to future releases with more features added. SecretsManager support, perhaps?
